pkgs/by-name/he/headphones-toolbox/package.nix

```
@@ -1,51 +1,77 @@
 {
+  cargo-tauri,
+  darwin,
+  fetchFromGitHub,
+  fetchYarnDeps,
   lib,
+  nix-update-script,
+  nodejs,
+  pkg-config,
+  rustPlatform,
   stdenv,
-  dpkg,
-  fetchurl,
-  autoPatchelfHook,
-  webkitgtk_4_0,
+  webkitgtk_4_1,
+  wrapGAppsHook3,
+  yarnConfigHook,
 }:
 
-stdenv.mkDerivation (finalAttrs: {
-  name = "headphones-toolbox";
-  version = "0.0.5";
+rustPlatform.buildRustPackage (rec {
+  pname = "headphones-toolbox";
+  version = "0.0.7";
+  tag = "test-tauri-v2-2";
 
-  src = fetchurl {
-    url = "https://github.com/ploopyco/headphones-toolbox/releases/download/app-v${finalAttrs.version}/ploopy-headphones-toolbox_${finalAttrs.version}_amd64.deb";
-    hash = "sha256-lWjmpybGcL3sbBng8zCTUtwYhlrQ6cCrKkhiu+g9MsE=";
+  src = fetchFromGitHub {
+    owner = "george-norton";
+    repo = pname;
+    rev = "${tag}";
+    hash = "sha256-X2HTEPxvBzbhfN1vqQVk81Qk1Z+EV+7/SpjZrDHv+fM=";
   };
 
-  nativeBuildInputs = [
-    dpkg
-    autoPatchelfHook
-  ];
+  offlineCache = fetchYarnDeps {
+    yarnLock = "${src}/yarn.lock";
+    hash = "sha256-Ln5U0KKsKm6ZLViZIWfBiBjm/mQNEIxaj4nTR55PcRg=";
+  };
 
-  buildInputs = [
-    webkitgtk_4_0
-  ];
+  useFetchCargoVendor = true;
+  cargoHash = "sha256-VgCxYYNBV45sTzouS5NE7nOUViPj0gJO7DSKlJSAT4U=";
+  cargoRoot = "src-tauri";
+  buildAndTestSubdir = cargoRoot;
 
-  installPhase = ''
-    runHook preInstall
+  nativeBuildInputs =
+    [
+      cargo-tauri.hook
+      nodejs
+      pkg-config
+      wrapGAppsHook3
+      yarnConfigHook
+    ]
+    ++ lib.optionals stdenv.hostPlatform.isDarwin [
+      darwin.DarwinTools # sw_vers
+    ];
 
-    mkdir -p $out/bin
-    mv usr/bin $out
-    mv usr/lib $out
-    mv usr/share $out
+  buildInputs =
+    lib.optionals stdenv.hostPlatform.isLinux [ webkitgtk_4_1 ]
+    ++ lib.optionals stdenv.hostPlatform.isDarwin [ darwin.apple_sdk.frameworks.WebKit ];
 
-    runHook postInstall
+  postInstall = lib.optionalString stdenv.hostPlatform.isDarwin ''
+    mkdir -p $out/bin
+    ln -s $out/Applications/Ploopy\ Headphones\ Toolbox.app/Contents/MacOS/${meta.mainProgram} $out/bin/${meta.mainProgram}
+    makeWrapper $out/{Applications/Ploopy\ Headphones\ Toolbox.app/Contents/MacOS,bin}/${meta.mainProgram}
   '';
 
-  meta = with lib; {
+  passthru.updateScript = nix-update-script { };
+
+  meta = {
     description = "UI for configuring Ploopy Headphones";
-    homepage = "https://github.com/ploopyco/headphones-toolbox/";
-    maintainers = with maintainers; [
+    homepage = "https://github.com/ploopyco/${meta.mainProgram}/";
```

```suggestion
    homepage = "https://github.com/ploopyco/headphones-toolbox/";
```
Please avoid interpolating with `meta.mainProgram` and `pname`, the first makes the derivation hash depend on `meta` which it shuold not and the latter is prone to break the override interface